utils/dataconvert/dataconvert.cpp

```diff
@@ -3427,9 +3430,12 @@ DataConvert::joinColTypeForUnion(datatypes::SystemCatalog::TypeHolderStd &unione
                 case datatypes::SystemCatalog::UDECIMAL:
                 case datatypes::SystemCatalog::UFLOAT:
                 case datatypes::SystemCatalog::UDOUBLE:
-                    unionedType.colDataType = datatypes::SystemCatalog::DOUBLE;
-                    unionedType.scale = 0;
-                    unionedType.colWidth = sizeof(double);
+                    if (!unionedType.isWideDecimalType())
+                    {
+                        unionedType.colDataType = datatypes::SystemCatalog::DOUBLE;
+                        unionedType.scale = 0;
+                        unionedType.colWidth = sizeof(double);
+                    }
```

Comment:
@drrtuy I believe you are referring to this block. Here I am checking if unionedType is a wide decimal, if so, then we don't reduce the type of unionedType to double. Same goes in a similar case block for long double.